test/tests/utils/test.helpers.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+define(['chai'],(chai)=>{
+    const safeBins = p5.prototype.safeBins;
+    let expect = chai.expect;
+
+    describe('safeBins', ()=>{
+        it('can handle negative input ',()=>{
+            let test =-Math.random()*2;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle value less than 16 ',()=>{
+            let test = 10;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle value greater than 1024',()=>{
+            let test = 1500;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle value other than power 2',()=>{
+            let test = 1500;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle value equal to 0',()=>{
+            let test = 0;
+            expect(safeBins(test)).to.equal(1024);
+        });
+        it('can handle strings',()=>{
+            let test = 'testString';    
+            expect(safeBins(test)).to.equal(1024);
+        });
+
+    });
+
+});
```

Comment:
this 🚫 means that the file doesn't end with a new line. I don't think it's a big deal but some do, so I think it's a best practice to end files with a new line :) https://thoughtbot.com/blog/no-newline-at-end-of-file

Reply:
are you talking about "\n" or what ?
 
i don't see this newline  in any of the test file in our library! should I use it everywhere ?

> this  means that the file doesn't end with a new line. I don't think it's a big deal but some do, so I think it's a best practice to end files with a new line :) https://thoughtbot.com/blog/no-newline-at-end-of-file